tests/charms/tls_certificates_interface/v1/test_tls_certificates_v1_requires.py

```diff
@@ -0,0 +1,594 @@
+        )
+
+        remote_app_relation_data = {
+            "certificates": [
+                {
+                    "ca": ca_certificate.decode(),
+                    "chain": ["a", "b"],
+                    "certificate_signing_request": certificate_signing_request.decode(),
+                    "certificate": certificate.decode(),
+                }
+            ]
+        }
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=remote_app,
+            key_values=remote_app_relation_data,
+        )
+
+        self.harness.charm.on.update_status.emit()
+
+        patch_certificate_expired.assert_not_called()
+
+    @patch(f"{BASE_CHARM_DIR}._on_certificate_expiring")
+    def test_given_certificate_expires_in_shorter_amount_of_time_than_expiry_notification_time_when_update_status_then_certificate_expiring_is_emitted(  # noqa: E501
+        self, patch_certificate_expiring
+    ):
+        remote_app = "tls-certificates-provider"
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=remote_app
+        )
+        hours_before_expiry = 8
+        private_key_password = b"whatever1"
+        ca_private_key_password = b"whatever2"
+        private_key = generate_private_key_helper(password=private_key_password)
+        ca_key = generate_private_key_helper(password=ca_private_key_password)
+        certificate_signing_request = generate_csr_helper(
+            private_key=private_key, private_key_password=private_key_password, subject="whatever"
+        )
+
+        ca_certificate = generate_ca_helper(
+            private_key=ca_key, private_key_password=ca_private_key_password, subject="whatever"
+        )
+
+        certificate = generate_certificate_helper(
+            ca=ca_certificate,
+            ca_key=ca_key,
+            csr=certificate_signing_request,
+            ca_key_password=ca_private_key_password,
+            validity=hours_before_expiry,
+        )
+
+        remote_app_relation_data = {
+            "certificates": [
+                {
+                    "ca": ca_certificate.decode(),
+                    "chain": ["a", "b"],
+                    "certificate_signing_request": certificate_signing_request.decode(),
+                    "certificate": certificate.decode(),
+                }
+            ]
+        }
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=remote_app,
+            key_values=remote_app_relation_data,
+        )
+
+        self.harness.charm.on.update_status.emit()
+
+        patch_certificate_expiring.assert_called()
+        args, _ = patch_certificate_expiring.call_args
+        event_data = args[0]
+        assert event_data.certificate == certificate.decode()
+        time_difference = datetime.fromisoformat(event_data.expiry) - datetime.utcnow()
+        assert (
+            (hours_before_expiry * 60 * 60) - 60
```

Comment:
I see this multiplication done a couple of times, can we move it to a constant?

```suggestion
            (hours_before_expiry * SECONDS_IN_ONE_HOUR) - 60
```

Reply:
Fixed